### PR TITLE
Add installation notes about using a python virtual environment

### PR DESCRIPTION
**Pull request by sivel**
This documentation change adds some information about running ansible from within a python virtual environment, listing out some specific caveats about the setup and configuration.

### CODE REVIEW

**Comment by mpdehaan** [comment]
I'm not sure I want to encourage virtualenv amongst the general userbase, who should think of Ansible as a general purpose tool not a language specific one. 

Possibly something we should put in the development section?

**Comment by sivel** [comment]
Sure, I can move it to the development section.

**Comment by sivel** [comment]
I'm not sure I see a great place to add it to the development section.  I think it should stay somewhat close to the 'Running From Source' section.

But if you have a specific recommendation of where to put it, I'd be open to suggestion.

**Comment by datasmid** [comment]
virtualenv isolates applications from each other and from the system's python. The way a python based product deals with dependencies (i.e. Python libraries and C bindings) determines acceptance by users, independent of your encouragement.

File: docsite/rst/intro_installation.rst [hunk 1]
```diff
@@ -217,6 +217,30 @@ Packaging Ansible or wanting to build a local package yourself, but don't want t
...
+.. _virtualenv:
+
+Using a virtual environment
++++++++++++++++++++++++++++
+
+Many users who wish to utilize a python virtual environment, will also utilize the previously mentioned instructions for running from source.  However, there is no reason you have to run from source in a virtual environment.i  Installing the latest release via pip as previously described will work just as well.
+
+Running ansible from within a python virtual environment requires several considerations.
+
+Many modules require additional python dependencies to be installed to operate. If the module dependencies are not installed globally, many modules will fail to operate as expected when executed with local_action or when specified to only run on localhost by another specification.  This is achievable utilizing the ``ansible_python_interpreter`` host variable. The localhost specification in your inventory will need to look similar to the following:
```
Comment (datasmid) [inline docsite/rst/intro_installation.rst:229]: Is there any way to know built-in module depencies dependencies before installation?

File: docsite/rst/intro_installation.rst [hunk 2]
```diff
@@ -217,6 +217,30 @@ Packaging Ansible or wanting to build a local package yourself, but don't want t
...
+
+.. code-block:: ini
+
+    [localhost]
+    localhost ansible_connection=local ansible_python_interpreter=/path/to/ansiblevenv/bin/python
```
Comment (datasmid) [inline docsite/rst/intro_installation.rst:234]: should'n this be inferred from the python interpreter that runs ansible? PEP-450...

**Comment by amenonsen** [comment]
I think it is sensible to add a brief note about running on virtualenv, but please, not with this wording (no offense intended, @sivel). Here's a suggested rewording (but see below):

You can run Ansible in a Python virtual environment either from source, as described above, or when installed by pip or using packages.

Many Ansible modules depend on Python modules, and if these are installed in the virtualenv but not globally, tasks that target `localhost` (e.g. with `local_action` or `hosts: localhost`) will fail. To use the virtual environment instead, set `ansible_python_interpreter` for localhost in your inventory, as shown below:

.. code-block:: ini
    [localhost]
    localhost ansible_connection=local ansible_python_interpreter=/path/to/ansiblevenv/bin/python

If you use `virtualenvwrapper <http://virtualenvwrapper.readthedocs.org>`_ you can use the `bin/postactivate` script to automatically source `hacking/env-setup` as shown below:

..code-block:: bash
    #!/bin/bash
    # This hook is run after this virtualenv is activated.
    source ${VIRTUAL_ENV}/ansible/hacking/env-setup -q

But please note @datasmid's questions earlier on this page. I do not have answers to those questions, but if there are easy answers, it would be nice to address them here. In particular, can we special-case finding python for -c local to use the same python that ansible was executed by? That would be preferable to documenting this... oddity.

File: docsite/rst/intro_installation.rst (hunk 2 above)
Comment (bcoca) [inline docsite/rst/intro_installation.rst:234]: you probably mean PEP-394 and no, as there are many systems that our users have to support that don't follow the proposal.

File: docsite/rst/intro_installation.rst (hunk 1 above)
Comment (bcoca) [inline docsite/rst/intro_installation.rst:229]: http://docs.ansible.com, the modules there have the requirements in the page.

**Comment by amenonsen** [comment]
This PR can be closed, I've pushed a revised version as #11689.

**Comment by gregdek** [comment]
Thanks @amenonsen!

**Pull request closed without merging**